DevRW/learning-reminder-backend
# #6 two factor authentication  &  Student CRUD

#### Introduction
Authenticate using two-factor authentication && CRUD of Student

#### Description of Task
- I have created the subscription modules
- I have created subscription service, helper and middlewares
- I have created student modules
- I created auth middlewares for subscribers

#### Test PR
- create subscription - send POST on **[localhost:5000/api/v1/subscriptions/create-subscription](localhost:5000/api/v1/subscriptions/create-subscription)**
`{
	"phoneNumber":"",
	"name":""
}`
after user will receive SMS code
- authenticate - POST on **[localhost:5000/api/v1/subscriptions/authenticate-subscriber](localhost:5000/api/v1/subscriptions/authenticate-subscriber)**
`{
	"phoneNumber":"",
	"code":""
}`
- user will receive token or auth error
- to login as subscriber -> POST **[localhost:5000/api/v1/subscriptions/login](localhost:5000/api/v1/subscriptions/login)**
`{
	"phoneNumber":""
}` after checking phone number, user will receive SMS verification code on his mobile phone 
- view my profile as subscriber - GET **[localhost:5000/api/v1/subscriptions/read-profile](localhost:5000/api/v1/subscriptions/read-profile)**
- add student as a parent **[localhost:5000/api/v1/students/create-student](localhost:5000/api/v1/students/create-student)**
`{
	"name":"kwitonda",
	"school":"gitisi",
	"classStudy":"1"
}`
- update student as a parent **[localhost:5000/api/v1/students/update-student/3](localhost:5000/api/v1/students/update-student/3)**
`{
	"name":"kwitonda jean paul",
	"school":"gitisi",
	"classStudy":"1"
}`
- delete student as a parent **[localhost:5000/api/v1/students/delete-student/3](localhost:5000/api/v1/students/delete-student/3)**

N.B: student actions should be acted by the authenticated parent
#### Author
[gratien](https://github.com/itsgracian)

#### Story ID
[#6](https://trello.com/c/XDwjOlDh/6-subscription-with-sms-auth)
